src/vconsole/vconsole-setup.c

```diff
@@ -82,7 +84,8 @@ static bool is_settable(int fd) {
          *
          * http://lists.freedesktop.org/archives/systemd-devel/2013-February/008573.html
          */
-        return r == 0 && IN_SET(curr_mode, K_XLATE, K_UNICODE);
+        return r < 0 ? -errno :
+               IN_SET(curr_mode, K_XLATE, K_UNICODE) ? 0 : (errno = EBUSY, -EBUSY);
```

Comment:
git grep ' errno = ' shows near 200 cases, so I assumed it was fine - especially if the caller has to act in the usual fashion (so it would expect errno to be sensibly set when the result is < 0 and errno may be passed down from some other function).

I changed is_settable() (and other functions that previously returned true/false) to follow < 0 / >= 0. Indirectly following Lennarts suggestions.